rest/diagnostic_doc_api.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2024-Present Couchbase, Inc.
+
+Use of this software is governed by the Business Source License included in
+the file licenses/BSL-Couchbase.txt.  As of the Change Date specified in that
+file, in accordance with the Business Source License, use of this software will
+be governed by the Apache License, Version 2.0, included in the file
+licenses/APL2.txt.
+*/
+
+package rest
+
+import (
+	"github.com/couchbase/sync_gateway/auth"
+	"github.com/couchbase/sync_gateway/db"
+)
+
+// HTTP handler for a GET of a document
+func (h *handler) handleGetDocChannels() error {
+	docid := h.PathVar("docid")
+
+	doc, err := h.collection.GetDocument(h.ctx(), docid, db.DocUnmarshalSync)
+	if err != nil {
+		return err
+	}
+	if doc == nil {
+		return kNotFoundError
+	}
+	resp := make(map[string][]auth.GrantHistorySequencePair, len(doc.Channels))
+
+	for _, chanSetInfo := range doc.SyncData.ChannelSet {
+		resp[chanSetInfo.Name] = append(resp[chanSetInfo.Name], auth.GrantHistorySequencePair{StartSeq: chanSetInfo.Start, EndSeq: chanSetInfo.End})
+	}
+	for _, hist := range doc.SyncData.ChannelSetHistory {
+		if _, ok := resp[hist.Name]; ok {
```

Comment:
```suggestion
		if _, ok := resp[hist.Name]; ok {
```

I don't think this is strictly nessesary. We'd always expect a ChannelSet entry if we have history, but I think it's worth omitting this check and always appending just in case.